### PR TITLE
Chore/add specs for allies controller

### PR DESCRIPTION
**Pull request by jmsardina**
This PR:
  - [x] introduces test coverage for the three actions in the
    AlliesController (`:index`, `:add`, `:remove`)
  - [x] introduces a pattern for shared_contexts and shared_examples
  - [ ] ~~sets default url host on `test.rb` to enable testing of ActionMailer integrations~~
  - [x] includes `require 'spec_helper'` in `.rspec` to allow for spec composition without adding `require 'spec_helper'` every time.
  - [ ] ~~adds `pry-nav` gem to facilitate debugging in the test and development environments~~

### CODE REVIEW

File: config/environments/test.rb [hunk 1]
```diff
@@ -11,6 +11,7 @@
   # just for the purpose of running a single test. If you are using a tool that
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
+  config.action_mailer.default_url_options = { :host => "http://localhost:3000" }
```
Comment (stellacotton) [inline config/environments/test.rb]: Looks like your fork may be a few commits behind master. I merged in a similar change to this file a few days ago. :)

File: Gemfile [hunk 2]
```diff
@@ -49,6 +49,7 @@ group :development, :test do
   gem 'better_errors',              '~> 2.1', '>= 2.1.1'
   gem 'annotate',                   '~> 2.6', '>= 2.6.10'
   gem 'pry-rails',                 '0.3.4'
+  gem 'pry-nav'
```
Comment (stellacotton) [inline Gemfile]: Is there a benefit to pry-nav over using byebug (which is already a dependency)? Looking at the pry-nav gem, it seems like they operate pretty similarly.

File: .rspec [hunk 3]
```diff
@@ -1 +1,2 @@
 --color
+--require spec_helper
```
Comment (stellacotton) [inline .rspec]: This change (+ the removals from the spec files) is so great. Thanks for doing it!

File: config/environments/test.rb (hunk 1 above)
Comment (jmsardina) [inline config/environments/test.rb]: Thanks @stellacotton!  I'll get my fork updated!

File: Gemfile (hunk 2 above)
Comment (jmsardina) [inline Gemfile]: Sorry about that!  I should have looked around the Gemfile first...
It seems `pry-nav` is not actively maintained, so this wouldn't have been an optimal addition.  I'll take it out of this PR!

**Comment by jmsardina** [comment]
@stellacotton I updated the branch with your repository and removed the unnecessary changes.

**Comment by stellacotton** [comment]
@jmsardina Awesome! PR looks good to me, I'm just checking in with @julianguyen on how she wants to handle merges.

**Pull request merged**